adapters/Mintegral/MintegralAdapter/GADMediationAdapterMintegral.m

```diff
@@ -116,53 +130,78 @@ + (GADVersionNumber)adapterVersion {
+    }
 }
 
-- (void)loadNativeAdForAdConfiguration:(GADMediationNativeAdConfiguration *)adConfiguration
-                     completionHandler:(GADMediationNativeLoadCompletionHandler)completionHandler {
-  _nativeAd = [[GADMAdapterMintegralNativeAdLoader alloc] init];
-  [_nativeAd loadNativeAdForAdConfiguration:adConfiguration completionHandler:completionHandler];
+- (void)loadNativeAdForAdConfiguration:(GADMediationNativeAdConfiguration *)adConfiguration completionHandler:(GADMediationNativeLoadCompletionHandler)completionHandler {
+    if([self isBiddingWithAdConfiguration:adConfiguration]){
+        _rtbNativeAd = [[GADMAdapterMintegralRTBNativeAdLoader alloc]init];
+        [_rtbNativeAd loadRTBNativeAdForAdConfiguration:adConfiguration completionHandler:completionHandler];
+    }else{
+        _nativeAd = [[GADMAdapterMintegralNativeAdLoader alloc]init];
+        [_nativeAd loadNativeAdForAdConfiguration:adConfiguration completionHandler:completionHandler];
+    }
 }
 
-- (void)loadRewardedAdForAdConfiguration:(GADMediationRewardedAdConfiguration *)adConfiguration
-                       completionHandler:
-                           (GADMediationRewardedLoadCompletionHandler)completionHandler {
-  _rewardedAd = [[GADMAdapterMintegralRewardedAdLoader alloc] init];
-  [_rewardedAd loadRewardedAdForAdConfiguration:adConfiguration
-                              completionHandler:completionHandler];
+- (void)loadRewardedAdForAdConfiguration:(GADMediationRewardedAdConfiguration *)adConfiguration completionHandler:(GADMediationRewardedLoadCompletionHandler)completionHandler {
+    if([self isBiddingWithAdConfiguration:adConfiguration]){
+        _rtbRewardedAd = [[GADMAdapterMintegralRTBRewardedAdLoader alloc]init];
+        [_rtbRewardedAd loadRTBRewardedAdForAdConfiguration:adConfiguration
+                                    completionHandler:completionHandler];
+    }else{
+        _rewardedAd = [[GADMAdapterMintegralRewardedAdLoader alloc]init];
+        [_rewardedAd loadRewardedAdForAdConfiguration:adConfiguration
+                                    completionHandler:completionHandler];
+    }
 }
 
-+ (void)setAdMobChannel {
-  // Set up the AdMob aggregation channel for server statistic collection purposes.
-  Class _class = NSClassFromString(@"MTGSDK");
-  SEL selector = NSSelectorFromString(@"setChannelFlag:");
-  NSString *pluginNumber = @"Y+H6DFttYrPQYcIBiQKwJQKQYrN=";
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Warc-performSelector-leaks"
-  if ([_class respondsToSelector:selector]) {
-    [_class performSelector:selector withObject:pluginNumber];
-  }
-#pragma clang diagnostic pop
+- (BOOL)isBiddingWithAdConfiguration:(GADMediationAdConfiguration *)adConfiguration {
+    //If there are bidResponse,it is considered bidding,otherwise it is waterfall.
+    if(adConfiguration.bidResponse && adConfiguration.bidResponse.length > 0){
+        return YES;
+    }else{
+        return NO;
+    }
+}
```

Comment:
I don't think we need this helper method, above we can just do: `if(adConfiguration.bidResponse)`.

Reply:
ok,I have modified.